HucaresServer/src/HucaresServer.DataAcquisition/CameraImageDownloading.cs

```diff
@@ -1,20 +1,59 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
 using HucaresServer.Storage.Helpers;
 
 namespace HucaresServer.DataAcquisition
 {
     public class CameraImageDownloading : ICameraImageDownloading
     {
-        private  ICameraInfoHelper _cameraInfoHelper;
+        // TODO: Take this value from a configuration file or somewhere else
+        private const string TemporaryStorageUrl = "/temporaryImages";
 
-        public CameraImageDownloading(ICameraInfoHelper cameraInfoHelper = null)
+        private ICameraInfoHelper _cameraInfoHelper;
+        private IImageSaver _imageSaver;
+        private IWebClientFactory _webClientFactory;
+
+        public CameraImageDownloading(ICameraInfoHelper cameraInfoHelper = null, IImageSaver imageSaver = null,
+            IWebClientFactory webClientFactory = null)
         {
             _cameraInfoHelper = cameraInfoHelper ?? new CameraInfoHelper();
+            _imageSaver = imageSaver ?? new LocalImageSaver(TemporaryStorageUrl);
+            _webClientFactory = webClientFactory ?? new CustomWebClientFactory();
         }
-        
+
         public int DownloadImagesFromCameraInfoSources(bool? isTrusted = null, DateTime? downloadDateTime = null)
         {
-            throw new System.NotImplementedException();
+            var cameraDataToDownload = _cameraInfoHelper.GetActiveCameras(isTrusted).ToList();
+            var imageSavingTasks = new List<Task>();
+
+            var datetime = downloadDateTime ?? DateTime.Now;
+
+            foreach (var cameraData in cameraDataToDownload)
+            {
+                imageSavingTasks.Add(Task.Factory.StartNew(
+                    () => DownloadAndSaveImage(cameraData.HostUrl, cameraData.Id, downloadDateTime)));
+            }
+
+            Task.WaitAll(imageSavingTasks.ToArray());
```

Comment:
This task is not awaited. As such this method will return before all the tasks have finished

Reply:
Awaited